proto/proto/note.proto

```diff
@@ -22,6 +22,7 @@ message NoteSyncRecord {
     digest.Digest note_id = 2;
     account.AccountId sender  = 3;
     fixed64 tag = 4;
+    uint32 note_type = 5;
```

Comment:
Should we use an [Enum](https://protobuf.dev/programming-guides/proto3/#enum) here?

Reply:
Wasn't familiar with it, took a look at the other types and did not see anything better so I used u32. When I added the enum originally I think I was hitting this: https://github.com/tokio-rs/prost/issues/927. But I can delete some of the codegen and it works fine.

````rust
  --> proto/src/generated/note.rs:60:41
   |
59 | #[derive(Eq, PartialOrd, Ord, Hash)]
   |          -- first implementation here
60 | #[derive(Clone, Copy, Debug, PartialEq, Eq, Hash, PartialOrd, Ord,...
   |                                         ^^ conflicting implementation for `note::NoteType`
   |
   = note: this error originates in the derive macro `Eq` (in Nightly builds, run with -Z macro-backtrace for more info)
````

Reply:
Never mind, it breaks the build obviously. Not sure how to fix this. Seems we generate the attributes on `build.rs` and this conflicts with some that prost is automatically adding?

```rust
 // Generate the stub of the user facing server from its proto file
    tonic_build::configure()
        .file_descriptor_set_path(&file_descriptor_path)
        .type_attribute(".", "#[derive(Eq, PartialOrd, Ord, Hash)]")
        .skip_protoc_run()
        .out_dir("src/generated")
        .compile_with_config(prost_config, protos, includes)
        .into_diagnostic()?;
```

Reply:
OK - I'm thinking of doing the following:

- Use `fixed32` in the message.
- Use the enum in the domain object.
- Create an issue to address this in the future.